src/main/src/main/java/org/geoserver/security/SecuredResourceNameChangeListener.java

```diff
@@ -51,37 +54,32 @@ public void handleAddEvent(CatalogAddEvent event) throws CatalogException {
 
     @Override
     public void handleRemoveEvent(CatalogRemoveEvent event) throws CatalogException {
-        String resourceWorkSpaceName = null;
-        CatalogInfo resourceInfo = event.getSource();
-        String resourceName = null;
-        if (resourceInfo instanceof LayerInfo) {
-            resourceWorkSpaceName =
-                    ((LayerInfo) resourceInfo).getResource().getStore().getWorkspace().getName();
-            resourceName = ((LayerInfo) resourceInfo).getName();
-        } else if (event.getSource() instanceof LayerGroupInfo) {
-            resourceWorkSpaceName = ((LayerGroupInfo) resourceInfo).getWorkspace().getName();
-            resourceName = ((LayerGroupInfo) resourceInfo).getName();
-        }
-
-        if (resourceWorkSpaceName != null && resourceInfo != null) {
-            // if a security rule exists for the layer whose name has been changed
-            if (layerHasSecurityRule(resourceWorkSpaceName, resourceName)) {
-                LOGGER.info(
-                        String.format(
-                                "Removing Security Rules for removed layer: %s", resourceName));
-
-                List<DataAccessRule> rulesToRemove = getDataAccessRule(resourceInfo, resourceName);
-                for (DataAccessRule r : rulesToRemove) {
-                    dao.removeRule(r);
-                }
-                try {
-                    dao.storeRules();
-                } catch (IOException e) {
-                    LOGGER.log(Level.SEVERE, e.getMessage(), e);
-                    throw new CatalogException(e);
-                }
+        Predicate<DataAccessRule> filter;
+        final String layerName;
+        final CatalogInfo eventSource = event.getSource();
+        if (eventSource instanceof LayerInfo) {
+            LayerInfo l = (LayerInfo) eventSource;
+            WorkspaceInfo ws = l.getResource().getStore().getWorkspace();
+            layerName = l.getName();
+            filter = filter(ws.getName()::equalsIgnoreCase, layerName::equalsIgnoreCase);
+        } else if (eventSource instanceof LayerGroupInfo) {
+            LayerGroupInfo lg = (LayerGroupInfo) eventSource;
+            WorkspaceInfo ws = lg.getWorkspace();
+            layerName = lg.getName();
+            if (ws == null) {
+                filter = filter(layerName::equalsIgnoreCase, r -> true);
```

Comment:
I was doubtful about this one, but yes, it seems a global layer group rule puts the group name in "root" and leaves the layer name empty. Maybe a comment about it would be useful.

Reply:
it is more obvious now, removed the `if`:
```
filter = layerGroupFilter(lg.getWorkspace(), lg.getName());
```
And [documented](https://github.com/geoserver/geoserver/pull/5620/files#diff-f76ab8b4ba5e8409fd7059659a76119a4f24a1dbe3aa8e0e6a45524951e8bdbaR171) the extracted method.